### ISSUE TITLE
Failed to Reproduce the Robust Encoder

### ISSUE THREAD

**Issue by lusen2000** [Fanghua-Yu/SUPIR#135]
Hi, I have read the Robust Encoder part of the paper and tried to reproduce it. I used the loss in the paper and the DF2K paired dataset which utilized Real-ESRGAN degradation. But it seems that the vae only output dark images after a few iters of training. Would you enlighten me with more details of this part?

**Comment by Try-to-Learn-Better** [comment]
> Hi, I have read the Robust Encoder part of the paper and tried to reproduce it. I used the loss in the paper and the DF2K paired dataset which utilized Real-ESRGAN degradation. But it seems that the vae only output dark images after a few iters of training. Would you enlighten me with more details of this part?

hello，sorry to bother you, but may I ask something about the training code? How do you finish train.py? As you see they didn't publish training codes on public , I also want to reproduce the robust encoder and I want to train it on more data. Will you upload your training code?

**Comment by Try-to-Learn-Better** [comment]
> Hi, I have read the Robust Encoder part of the paper and tried to reproduce it. I used the loss in the paper and the DF2K paired dataset which utilized Real-ESRGAN degradation. But it seems that the vae only output dark images after a few iters of training. Would you enlighten me with more details of this part?

Hello,I also find after denoise_encoder and decoder, the input turns out dark output. Do you solve the problem?

**Comment by lusen2000** [comment]
> > Hi, I have read the Robust Encoder part of the paper and tried to reproduce it. I used the loss in the paper and the DF2K paired dataset which utilized Real-ESRGAN degradation. But it seems that the vae only output dark images after a few iters of training. Would you enlighten me with more details of this part?
> 
> Hello,I also find after denoise_encoder and decoder, the input turns out dark output. Do you solve the problem?

Yes, you should also use the regular loss of VAE combined with the robust-vae loss, if you only use the latter one, the image will get blur or dark.
